packages/angular_devkit/build_angular/src/webpack/configs/typescript.ts

```diff
@@ -13,9 +13,95 @@ import {
   AngularCompilerPlugin,
   AngularCompilerPluginOptions,
   NgToolsLoader,
-  PLATFORM
+  PLATFORM,
+  ivy,
 } from '@ngtools/webpack';
 import { WebpackConfigOptions, BuildOptions } from '../../utils/build-options';
+import { CompilerOptions } from '@angular/compiler-cli';
+import { RuleSetLoader } from 'webpack';
+
+function canUseIvyPlugin(wco: WebpackConfigOptions): boolean {
+  // Can only be used with Ivy
+  if (!wco.tsConfig.options.enableIvy) {
+    return false;
+  }
+
+  // Allow new ivy build system via environment variable ('NG_BUILD_IVY_EXPERIMENTAL=1')
+  // TODO: Remove this section and adjust warnings below once the Ivy plugin is the default
+  const flag = process.env['NG_BUILD_IVY_EXPERIMENTAL'];
+  if (flag === undefined || flag === '0' || flag.toLowerCase() === 'false') {
+    return false;
+  }
+
+  // Allow fallback to legacy build system via environment variable ('NG_BUILD_IVY_LEGACY=1')
+  // TODO: Enable this section once the Ivy plugin is the default
+  // const flag = process.env['NG_BUILD_IVY_LEGACY'];
+  // if (flag !== undefined && flag !== '0' && flag.toLowerCase() !== 'false') {
+  //   wco.logger.warn(
+  //     '"NG_BUILD_IVY_LEGACY" environment variable detected. Using legacy Ivy build system.',
+  //   );
+
+  //   return false;
+  // }
+
+  // Lazy modules option uses the deprecated string format for lazy routes which is not supported
+  if (wco.buildOptions.lazyModules && wco.buildOptions.lazyModules.length > 0) {
+    wco.logger.warn(
+      '"lazyModules" option is deprecated and not supported by the experimental Ivy build system. ' +
+        'Using existing Ivy build system.'
```

Comment:
I think "existing" is confusing here, how about "original" (to avoid calling it "legacy" while it's not enabled by default). Same for the message below.